mediators/datamapper/components/org.wso2.carbon.gateway.mediators.datamapper.engine/src/main/java/org/wso2/carbon/gateway/mediators/datamapper/engine/input/readers/XMLInputReader.java

```diff
@@ -0,0 +1,588 @@
+            writeArrayEndElement();
+        }
+
+        if (isObject) {
+            writeObjectEndElement();
+        }
+
+        if (isArrayElement) {
+            return nameSpaceLocalName;
+        }
+        return null;
+    }
+
+    /**
+     * This method is used to get the namespace URI of an XML element (OMElement).
+     *
+     * @param omElement
+     * @return Namespace URI of the given OMElement, if there is no Namespace return an empty String
+     */
+    private String getNameSpaceURI(OMElement omElement) {
+        String nameSpaceURI = "";
+        if (omElement.getNamespace() != null) {
+            nameSpaceURI = omElement.getNamespace().getNamespaceURI();
+        }
+        return nameSpaceURI;
+    }
+
+    /**
+     * This method is used to get the namespace URI of an XML attribute (OMAttribute).
+     *
+     * @param omAttribute
+     * @return Namespace URI of the given OMAttribute, if there is no Namespace return an empty String
+     */
+    private String getNameSpaceURI(OMAttribute omAttribute) {
+        String nameSpaceURI = "";
+        if (omAttribute.getNamespace() != null) {
+            nameSpaceURI = omAttribute.getNamespace().getNamespaceURI();
+        }
+        return nameSpaceURI;
+    }
+
+    /**
+     * This method writes attribute elements into the JSON input message.
+     *
+     * @param jsonSchemaMap current level JSON Schema
+     * @throws JSException
+     * @throws SchemaException
+     * @throws ReaderException
+     * @throws IOException
+     * @throws InvalidPayloadException
+     */
+    private void writeAttributes(Map jsonSchemaMap)
+            throws JSException, SchemaException, ReaderException, IOException, InvalidPayloadException {
+
+        /* currently processing attribute element and its parameters*/
+        String attributeType;
+        String attributeFieldName;
+        String attributeLocalName;
+        String attributeNSURI;
+        String attributeQName;
+        OMAttribute omAttribute;
+
+        /** Writing next attributes to the JSON message */
+        while (it_attr.hasNext()) {
+            omAttribute = it_attr.next();
+
+            /* skip if the attribute name contains "XMLNS" */
+            attributeLocalName = omAttribute.getLocalName();
+            if (attributeLocalName.contains(XMLNS))
+                continue;
+
+            attributeNSURI = this.getNameSpaceURI(omAttribute);
+            attributeFieldName = getAttributeFieldName(attributeLocalName, attributeNSURI);
+            attributeQName = getAttributeQName(omAttribute.getNamespace(), attributeLocalName);
+
+            /* get the type of the attribute element */
+            attributeType = getElementType(jsonSchemaMap, attributeQName);
+            if (NULL_ELEMENT_TYPE.equals(attributeType)) {
+                 /* Check whether the input payload has empty tags. */
+                log.warn("Attribute name not found : " + attributeQName);
```

Comment:
can this warning message be improved more so anyone whose looking at the log have more context into what went wrong?